### PR TITLE
Refactoring, and only one, true, RTTI dynamic call.

### PR DESCRIPTION
**Pull request by dkorolev**
Finally:
* Retired the old, O(n), `Bricks/rtti/dispatcher.h`, and
* Regenerated `README.md`-s in Bricks and Blocks.

Also:
* Made this change as RipCurrent-free as possible.

### CODE REVIEW

**Comment by dkorolev** [comment]
Safe to say this code passes CI -- the red one is Travis' native flakiness.

File: RipCurrent/test.cc [hunk 1]
```diff
@@ -484,6 +490,52 @@ TEST(RipCurrent, CustomTypesFlow) {
...
+    EXPECT_EQ("OK\n", HTTP(HEAD(base_url)).body);
+    EXPECT_EQ("OK\n", HTTP(POST(base_url, "OK")).body);
+  }
+  HTTPRoutesScope scope;
+};
+#define RCHTTPAcceptor(...) RIPCURRENT_MACRO(RCHTTPAcceptor, __VA_ARGS__)
+
+RIPCURRENT_NODE(RCHTTPResponder, RequestContainer, void) {
+  std::vector<std::string>& requests;
+  RCHTTPResponder(std::vector<std::string> & requests) : requests(requests) {}
```
Comment (mzhurovich) [inline RipCurrent/test.cc]: Extra space: `std::vector<std::string> &`

File: TypeSystem/types.h [hunk 2]
```diff
@@ -41,6 +41,13 @@ struct CurrentSuper {
   virtual ~CurrentSuper() = default;
 };
 
+// For `unique_ptr<>`-s.
+struct CurrentSuperDeleter {
+  void operator()(CurrentSuper* ptr) {
+    delete ptr;
+  }
+};
```
Comment (mzhurovich) [inline TypeSystem/types.h]: The only place `CurrentSuperDeleter` is mentioned in this PR is this file. Is it for future use or some legacy?

File: Blocks/HTTP/request.h [hunk 3]
```diff
@@ -51,7 +51,7 @@ constexpr static auto HasRespondViaHTTP(int)
 }
 
 // The only parameter to be passed to HTTP handlers.
-struct Request final {
+struct Request final : CurrentSuper {
```
Comment (mzhurovich) [inline Blocks/HTTP/request.h]: Why do you need it?

Comment (dkorolev) [inline Blocks/HTTP/request.h]: To be able to `std::move()` a `Request` itself, without a wrapper class, contained in the `std::unique_ptr<CurrentSuper, CurrentSuperDeleter>` which is within the MM[P]Q of RipCurrent.

In other words, a RipCurrent artifact of this code, sorry it's part of this PR.

File: RipCurrent/test.cc (hunk 1 above)
Comment (dkorolev) [inline RipCurrent/test.cc]: a58d5c4

File: Bricks/README.md [hunk 4]
```diff
@@ -15,18 +15,13 @@ HTTP server and client, JSON and binary serialization, visualization and other c
   [tar.gz](https://github.com/C5T/Current/archive/v1.0.tar.gz),
   [zip](https://github.com/C5T/Current/archive/v1.0.zip).
 
-  HTTP server and client API-s, RapidJSON for JSON serialization, [gnuplot](http://www.gnuplot.info/) and [plotutils](http://www.gnu.org/software/plotutils/) ports, file system and string manipulation methods, a command line flags parsing tool and a header-only port of [GoogleTest](http://code.google.com/p/googletest/).
+  HTTP server and client API-s, [Cereal](http://uscilab.github.io/cereal/) for binary and JSON serialization, [gnuplot](http://www.gnuplot.info/) and [plotutils](http://www.gnu.org/software/plotutils/) ports, file system and string manipulation methods, a command line flags parsing tool and a header-only port of [GoogleTest](http://code.google.com/p/googletest/).
```
Comment (dkorolev) [inline Bricks/README.md]: I went on to fix it too upon noticing, only to realize these are the release notes of Bricks 1.0. :-) Suggest we keep it for now, and retire the whole section once we release Current 1.0.

**Comment by dkorolev** [comment]
@mzhurovich, thanks for the comments and PTAL.

**Pull request merged**